docs/_static/nuts-network.yaml

```diff
@@ -0,0 +1,131 @@
+openapi: "3.0.0"
+info:
+  title: Nuts network API spec
+  description: API specification for RPC services available at the nuts-network
+  version: 0.1.0
+  license:
+    name: GPLv3
+paths:
+  /api/document:
+    get:
+      summary: "Lists the documents on the log"
+      operationId: "listDocuments"
+      tags:
+        - documents
+      responses:
+        "200":
+          description: "Successfully listed the documents"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Document'
+    post:
+      summary: "Adds a document to the document log"
+      operationId: "addDocument"
+      tags:
+        - documents
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DocumentWithContents'
+      responses:
+        "201":
+          description: "Document was successfully added to the document log"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Document'
+        "409":
+          description: "Document with that hash already exists"
+          content:
+            text/plain:
+              example: "Document already exists"
+        "400":
+          description: "Supplied document is invalid"
+          content:
+            text/plain:
+              example: "Document hash mismatch"
+  /api/document/{hash}:
+    parameters:
+      - name: hash
+        in: path
+        description: "Hex-encoded hash of the document"
+        required: true
+        example: "37076f2cbe014a109d79b61ae9c7191f4cc57afc"
+        schema:
+          type: string
+    get:
+      summary: "Retrieves a document"
+      operationId: "getDocument"
+      tags:
+        - documents
+      responses:
+        "200":
+          description: "Document is known in the document log"
+        "404":
+          description: "Document wasn't found in the document log"
+  /api/document/{hash}/contents:
+    parameters:
+      - name: hash
+        in: path
+        description: "Hex-encoded hash of the document"
+        required: true
+        example: "37076f2cbe014a109d79b61ae9c7191f4cc57afc"
+        schema:
+          type: string
+    get:
+      summary: "Gets the document contents"
+      operationId: "getDocumentContents"
+      tags:
+        - documents
+      responses:
+        "200":
+          description: "Document found (with contents) and returned."
+          content:
+            application/octet-stream:
```

Comment:
The network doesn't really care what type of data is stored in the documents and it might even be binary (or plain text, JSON or a Bash script), so the API can't really tell at this point what content type it should return.

However, I do see that it could be useful from a caller's perspective. Problem is that documents don't specify their content types. Is that something we should add, from your perspective?

Reply:
Something to discuss, cause it's also an architectural thing. I think only business logic is required to read it?

Reply:
Yes, since the network doesn't care about what it's transmitting. Business applications could derive it from the document type, I think that would be good enough for most cases (at least consents and the registry) but maybe in the future we need more metadata. When you have a theoretical document type 'photos.kitten' you could have content-type image/jpg, image/png, etc.